File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"
#include <set>

namespace pangram {

bool is_pangram(const std::string& sentence) {
    std::set<char> letters;
    for (auto c: sentence)
        if (isalpha(c))
            letters.insert(tolower(c));
    return letters.size() == 26;
}

}  // namespace pangram

File: solutions/haskell/pangram/1/src/Pangram.hs
module Pangram (isPangram) where
import Data.List
import Data.Char

letters = "abcdefghijklmnopqrstuvwxyz"

isPangram :: String -> Bool
isPangram text = sort (nub (filtered (map toLower text))) == letters

filtered :: String -> String
filtered text = filter (\c -> elem c letters) text